DDOT_module/child_classes/ddot_marginal_analyzer_vanilla.py

```diff
@@ -0,0 +1,396 @@
+            states_after_averaging}
+
+        qubits_to_be_averaged_over = list(set(all_neighbors).difference(set(qubits_to_be_left)))
+        qubits_to_be_averaged_over_mapped = [reversed_enumerated[q] for q in
+                                             qubits_to_be_averaged_over]
+
+        for neighbors_state, conditional_noise_matrix in matrices_cluster.items():
+            list_string_neighbors = list(copy.deepcopy(neighbors_state))
+
+            list_string_neighbors = list(np.delete(list_string_neighbors,
+                                                   qubits_to_be_averaged_over_mapped))
+
+            string_neighbors = ''.join(list_string_neighbors)
+
+            # TODO FBM: make sure this commented check is not needed anymore
+            # if string_neighbors == '' or string_neighbors:
+            #     try:
+            #         averaged_matrices_cluster['averaged'] +=
+            #         conditional_noise_matrix / averaging_normalization
+            #     except(KeyError):
+            #         averaged_matrices_cluster['averaged'] =
+            #         conditional_noise_matrix/averaging_normalization
+            # else:
+            averaged_matrices_cluster[
+                string_neighbors] += conditional_noise_matrix / averaging_normalization
+
+        return averaged_matrices_cluster
+
+    def _compute_noise_matrix_averaged(self,
+                                       subset: List[int]) -> np.ndarray:
+        """Noise matrix for subset of qubits, averaged over all other qubits
+           Defaultly takes data from self._marginals_dictionary. If data is not present, then it
+           calculates marginals for given subset
+           and updates the class's property self.marginals_dictionary
+        """
+
+        # TODO: Perhaps add possibility of using existing marginals for bigger subset that includes
+        # target subset
+
+        subset_key = 'q' + 'q'.join([str(s) for s in subset])
+        marginals_dictionary_ddot = self._marginals_dictionary
+
+        # TODO FBM: testing some issue with normalization
+        testing_method = False
+
+        # TODO FBM: refactor defaultdict usage here
+        marginal_dict_now = defaultdict(float)
+
+        for input_state_bitstring, dictionary_marginals_now in marginals_dictionary_ddot.items():
+            input_marginal = ''.join([input_state_bitstring[x] for x in subset])
+
+            if subset_key not in dictionary_marginals_now.keys():
+                self.compute_marginals(input_state_bitstring, [subset])
+
+            marginal_dict_now[input_marginal] += dictionary_marginals_now[subset_key]
+
+            # TODO FBM: this perhaps is incosistend method, make sure.
+            if testing_method:
+                for key_small in marginal_dict_now.keys():
+                    marginal_dict_now[key_small] *= 1 / np.sum(marginal_dict_now[key_small])
```

Comment:
```suggestion
                    marginal_dict_now[key_small] /=  np.sum(marginal_dict_now[key_small])
```
Also - is there a risk of zero division?